spec/models/line_item_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe LineItem, type: :model do
+  describe "associations" do
+    it { is_expected.to belong_to(:cart) }
+    it { is_expected.to belong_to(:book) }
+  end
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:book) }
```

Comment:
Nice! I didn't know there was an expect helper for this. I'll have to use that in the project I'm working on.